extension/services/ExtensionMetadataReader.cfc

```diff
@@ -0,0 +1,327 @@
+component accessors=true {
+
+	property name="s3root"            type="string" default="";
+	property name="extensionMeta"     type="query";
+	property name="extensionVersions" type="struct";
+
+	variables._simpleCache = {};
+
+	function loadMeta() {
+		var meta           = _readExistingMetaFileFromS3();
+		var existingByFile = _mapExtensionQueryByFilename( meta );
+		var lexFiles       = _listLexFilesFromBucket();
+		var metaChanged    = false;
+
+		for( var lexFile in lexFiles ) {
+			var isNewToUs = !StructKeyExists( existingByFile, lexFile.name )
+			if ( isNewToUs ) {
+				_addLexFile( lexFile.name, meta );
```

Comment:
this could fail for any number of reasons, one bad egg shouldn't block other extensions from being published

Reply:
Yes, there is try/catch within addLexFile